posthog/warehouse/api/table.py

```diff
@@ -266,3 +272,112 @@ def refresh_schema(self, request: request.Request, *args: Any, **kwargs: Any) ->
+
+        file = request.FILES["file"]
+        table_name = request.data.get("name", file.name)
+        file_format = request.data.get("format", "CSVWithNames")
+
+        # Validate table name format
+        if not re.match(r"^[a-zA-Z_][a-zA-Z0-9_]*$", table_name) or not re.match(
+            r"^[a-zA-Z_][a-zA-Z0-9_]*$", file.name
+        ):
+            return response.Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": "Table names must start with a letter or underscore and contain only alphanumeric characters or underscores."
+                },
+            )
+
+        # Validate table name
+        team_id = self.team_id
+        table = None
+        table_query = DataWarehouseTable.objects.exclude(deleted=True).filter(team_id=team_id, name=table_name)
+        if table_query.exists():
+            table = table_query.first()
+
+        if file.size > 52428800:  # 50MB in bytes
+            return response.Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"message": f"File size exceeds maximum allowed size of 50MB"},
+            )
+
+        # Create the table record
+        try:
+            # Create credential if object storage is available
+            credential = None
+            if hasattr(settings, "AIRBYTE_BUCKET_KEY") and hasattr(settings, "AIRBYTE_BUCKET_SECRET"):
+                credential = get_or_create_datawarehouse_credential(
+                    team_id=team_id,
+                    access_key=settings.AIRBYTE_BUCKET_KEY,
+                    access_secret=settings.AIRBYTE_BUCKET_SECRET,
+                )
+
+            # Create the table if it doesn't exist, otherwise use existing one
+            if table is None:
+                table = DataWarehouseTable.objects.create(
+                    team_id=team_id,
+                    name=table_name,
+                    format=file_format,
+                    created_by=request.user,
+                    credential=credential,  # type: ignore
+                )
+
+            # Generate URL pattern and store file in object storage
+            if credential and settings.DATAWAREHOUSE_BUCKET:
+                s3 = boto3.client(
+                    "s3", aws_access_key_id=credential.access_key, aws_secret_access_key=credential.access_secret
+                )
+                s3.upload_fileobj(file, settings.DATAWAREHOUSE_BUCKET, f"dlt/managed/team_{team_id}/{file.name}")
+
+                # Set the URL pattern for the table
+                table.url_pattern = f"https://{settings.AIRBYTE_BUCKET_DOMAIN}/dlt/managed/team_{team_id}/{file.name}"
+                table.format = file_format
+
+                if table.credential is None:
+                    table.credential = credential
+
+                # Try to determine columns from the file
+                table.columns = table.get_columns()
```

Comment:
I dont think this should really break the 60 second request timeout rule, but a heads up incase things do weirdly break cos of timeouts - this will likely be the culprit